File: main/src/main/java/com/sedmelluq/discord/lavaplayer/container/MediaContainerRegistry.java
package com.sedmelluq.discord.lavaplayer.container;

import java.util.List;

public class MediaContainerRegistry {
  public static final MediaContainerRegistry DEFAULT_REGISTRY = new MediaContainerRegistry(MediaContainer.asList());

  private final List<MediaContainerProbe> probes;

  public MediaContainerRegistry(List<MediaContainerProbe> probes) {
    this.probes = probes;
  }

  public MediaContainerProbe find(String name) {
    for (MediaContainerProbe probe : probes) {
      if (name.equals(probe.getName())) {
        return probe;
      }
    }

    return null;
  }

  public List<MediaContainerProbe> getAll() {
    return probes;
  }

  public static MediaContainerRegistry extended(MediaContainerProbe... additional) {
    List<MediaContainerProbe> probes = MediaContainer.asList();

    for (MediaContainerProbe probe : additional) {
      probes.add(probe);
    }

    return new MediaContainerRegistry(probes);
  }
}

File: main/src/main/java/lavaplayer/container/MediaContainerRegistry.kt
package lavaplayer.container

open class MediaContainerRegistry(vararg val probes: MediaContainerProbe) {
    companion object {
        @JvmField
        val DEFAULT_REGISTRY = MediaContainerRegistry(MediaContainer.asList())

        fun extended(vararg additional: MediaContainerProbe): MediaContainerRegistry {
            val probes = MediaContainer.asList()
            probes.addAll(probes.intersect(additional.toList()))
            return MediaContainerRegistry(probes)
        }
    }

    constructor(probes: List<MediaContainerProbe>) : this(*probes.toTypedArray())

    fun find(name: String): MediaContainerProbe? =
        probes.find { it.name == name }
}
